File: Jaewon/Level.1/2) 모의고사/2.js
function solution(answers) {
    var answer = [];
    var sampleB = [2, 1, 2, 3, 2, 4, 2, 5];
    var sampleC = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5];
  
    var abandoner = 3;
    var temp = [];
    for(let i = 0; i < 3; i++){
        temp[i] = 0;
    }
    
    for(let i=0; i<answers.length; i++){
        if( answers[i] === i%5+1 ){
            temp[0] = temp[0] + 1;
        }
        if( answers[i] === sampleB[i%sampleB.length]){
            temp[1] = temp[1] + 1;
        }
        if( answers[i] === sampleC[i%sampleC.length]){
            temp[2] = temp[2] + 1;
        }
    }
   
    var max = Math.max.apply(null, temp);
    for(let i = 0; i < abandoner; i++){
        if( max === temp[i]){
            answer.push(i+1);
        }
    }
    
    return answer;
}

File: Jaewon/Level.1/2) 모의고사/2.py
def solution(answers):
    answer = []
    sampleB = [2, 1, 2, 3, 2, 4, 2, 5];
    sampleC = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5];
    
    abandoner = 3;
    temp = [];
    
    for i in range(abandoner):
        temp.append(0)
    
    for idx, select in enumerate(answers):
        if select == idx%5+1:
            temp[0] += 1
        if select == sampleB[idx%len(sampleB)]:
            temp[1] += 1
        if select == sampleC[idx%len(sampleC)]:
            temp[2] += 1
            
    for idx, result in enumerate(temp):
        if result == max(temp):
            answer.append(idx+1)
            
    return answer